internal/server/server_authz_checks.go

```diff
@@ -2,16 +2,66 @@ package server
 
 import (
 	"context"
+	"crypto/hmac"
+	"crypto/sha256"
+	"crypto/subtle"
+	"encoding/hex"
+	"io"
 	"net/http"
 
 	"github.com/kopia/kopia/internal/auth"
 )
 
+const kopiaSessionCookie = "Kopia-Session-Cookie"
+
+func (s *Server) generateCSRFToken(sessionID string) string {
+	h := hmac.New(sha256.New, s.authCookieSigningKey)
+	io.WriteString(h, sessionID) //nolint:errcheck
+
+	return hex.EncodeToString(h.Sum(nil))
+}
+
+func (s *Server) validateCSRFToken(r *http.Request) bool {
+	if s.options.DisableCSRFTokenChecks {
+		return true
+	}
+
+	ctx := r.Context()
+	path := r.URL.Path
+
+	sessionCookie, err := r.Cookie(kopiaSessionCookie)
+	if err != nil {
+		log(ctx).Warnf("missing or invalid session cookie for %v", path)
+
+		return false
+	}
```

Comment:
nit; It does simplify a few things if `validateCSRFToken` returns a boolean always. However, in this particular scenario it'll eat some precious debugging information when systems are under load/memory-pressure.

I'd be desirable if `validateCSRFToken` returns false only for `ErrNoCookie`. For anything else `validateCSRFToken` should probably wrap the error with some context and return it along with `false`.

Reply:
I don't see why this matters, CSRF token that is missing, malformed or invalid should have the same effect, right?

Reply:
missing would throw `ErrNoCookie`

malformed / invalid would be something we probably should try not to return a boolean for, that in my head is indicative of a third state which can be "I don't know".

not a deal breaker though, so please don't block on this comment